app/src/main/java/com/swent/skillswap/model/chat/ChatRepositoryFirestore.kt

```diff
@@ -97,4 +121,49 @@ class ChatRepositoryFirestore(private val db: FirebaseFirestore) : ChatRepositor
         /** Clean up listener on flow cancellation */
         awaitClose { registration.remove() }
     }
+
+    override suspend fun getChatsOfCurrentUser(relatedPostType: PostType): List<Chat> {
+
+        val currentUserId =
+            try {
+                Firebase.auth.currentUser?.uid ?: throw Exception("No authenticated user found")
+            } catch (e: Exception) {
+                throw Exception("Error while retrieving current user ID: ${e.message}")
+            }
```

Comment:
I think the try/catch block is redundant here since Firebase.auth.currentUser?.uid is null when no user is found and if any  error occurs with Firebase.auth it is already handled internally by firebase auth libraries (also if the firebase auth instance of the app is failling then the app will most likely not even run )